### ISSUE TITLE
Types highlighting support

### ISSUE THREAD

**Issue by cmpute** [guyskk/language-cython-for-vscode#5]
Hope that the type name can be highlighted like in C++. This might be quite different from python highlighting since this requires to detect type names. For example the types in following statement:
```cython
cdef <typename> <obj>
cdef cppclass <typename>[<template>]
cdef struct <typename>
def/cdef/cpdef <return type> <function>(<arg type1> <arg name1>, <arg type2> <arg name2>, ...)
```

**Comment by guyskk** [comment]
Do you means user defined type name? if yes then it's hard to support.
otherwise we can treat them as keywords and it's easy, you can search keywords like `print` in tmLanguage file and append type names to it.

**Comment by cmpute** [comment]
@guyskk Yes, I mean user defined type. It may need some semantic analysis :[
